File: FamilyAppFlutter/android/buildSrc/src/main/java/dev/flutter/FlutterGradlePlugin.java
package dev.flutter;

import org.gradle.api.Plugin;
import org.gradle.api.Project;

public class FlutterGradlePlugin implements Plugin<Project> {
    @Override
    public void apply(Project target) {
        target.getLogger().lifecycle("Applying stub Flutter Gradle plugin for Android-only builds");
        target.getExtensions().getExtraProperties().set("flutterEmbeddingVersion", "stub");
    }
}

File: FamilyAppFlutter/android/buildSrc/src/main/kotlin/dev/flutter/FlutterGradlePlugin.kt
package dev.flutter

import org.gradle.api.Plugin
import org.gradle.api.Project

/**
 * Minimal stub of the Flutter Gradle plugin that keeps the Android module buildable
 * when the Flutter SDK is not available in the environment.
 */
class FlutterGradlePlugin : Plugin<Project> {
    override fun apply(target: Project) {
        target.logger.lifecycle("Applying stub Flutter Gradle plugin for Android-only builds")

        target.plugins.withId("com.android.application") {
            target.extensions.extraProperties.set("flutterEmbeddingVersion", "stub")
        }
    }
}
